### PR TITLE
fix: Correctly handle log level configuration

### PR DESCRIPTION
**Pull request by janbuchar**
- The `CRAWLEE_VERBOSE` and `CRAWLEE_LOG_LEVEL` variables are now respected by crawlee
- `CrawleeLogFormatter` doesn't alter the record being formatted anymore
- SDK counterpart coming soon

### CODE REVIEW

File: src/crawlee/configuration.py [hunk 1]
```diff
@@ -51,14 +51,14 @@ class Configuration(BaseSettings):
     ] = False
 
     log_level: Annotated[
-        int,
+        Literal['DEBUG', 'INFO', 'WARNING', 'ERROR'],
```
Comment (vdusek) [inline src/crawlee/configuration.py:54]: `NOTSET`? `CRITICAL`?

File: src/crawlee/basic_crawler/_basic_crawler.py [hunk 2]
```diff
@@ -203,20 +203,14 @@ def __init__(
...
-            root_logger.addHandler(handler)
-            root_logger.setLevel(logging.INFO if not sys.flags.dev_mode else logging.DEBUG)
+            configure_logger(root_logger, self._configuration, remove_old_handlers=True)
 
             # Silence HTTPX logger
             httpx_logger = logging.getLogger('httpx')
-            httpx_logger.setLevel(logging.WARNING if not sys.flags.dev_mode else logging.INFO)
+            httpx_logger.setLevel(
+                logging.DEBUG if get_configured_log_level(self._configuration) <= logging.DEBUG else logging.WARNING
+            )
```
Comment (vdusek) [inline src/crawlee/basic_crawler/_basic_crawler.py:213]: Not so important, but why did you change this and how does it differ from the previous version?

File: src/crawlee/_log_config.py [hunk 3]
```diff
@@ -31,6 +35,44 @@
...
+
+    if 'log_level' in configuration.model_fields_set:
+        if configuration.log_level == 'DEBUG':
+            return logging.DEBUG
+        if configuration.log_level == 'INFO':
+            return logging.INFO
+        if configuration.log_level == 'WARNING':
+            return logging.WARNING
+        if configuration.log_level == 'ERROR':
+            return logging.ERROR
```
Comment (vdusek) [inline src/crawlee/_log_config.py:49]: why not support all possible log levels?

**Review by vdusek — CHANGES_REQUESTED** [review]

File: src/crawlee/configuration.py (hunk 1 above)
Comment (janbuchar) [reply src/crawlee/configuration.py:54]: These are what Javascript crawlee supports. Notset doesn't sound like a great idea to me here, but I'm open to adding `CRITICAL`.

File: src/crawlee/basic_crawler/_basic_crawler.py (hunk 2 above)
Comment (janbuchar) [reply src/crawlee/basic_crawler/_basic_crawler.py:213]: It uses a helper function to determine the log level instead of repeating an incomplete condition all over the place. In the default case, it should behave the same as before.

File: src/crawlee/_log_config.py (hunk 3 above)
Comment (janbuchar) [reply src/crawlee/_log_config.py:49]: Less is more :shrug: Same as the comment higher up, I'd consider adding `CRITICAL`. Am I missing something?

Comment (vdusek) [reply src/crawlee/_log_config.py:49]: ok, adding just `CRITICAL` si good, thanks

**Review by vdusek — APPROVED** [review]
just the CRITICAL thing, otherwise LGTM :rocket:
